### DISCUSSION TITLE
Disable/hide fields based on role

### DISCUSSION THREAD

**Discussion by mhdaouas** [directus/directus#14691] [Archive]
I would like to disable/hide fields based on the role of the user who accesses the detail/content page (form page) and the status (or any other param, it doesn't matter). How can I proceed?

**Comment by br41nslug** [comment]
You can achieve this using field conditions (Settings > Data Model > COLLECTION > FIELD > Conditions) https://docs.directus.io/configuration/data-model/#conditions

**Reply by mhdaouas** [reply]
I don't think I can get the current user role there or am I wrong?

**Reply by br41nslug** [reply]
You can use the dynamic variable `$CURRENT_ROLE` for that https://docs.directus.io/reference/filter-rules/#dynamic-variables

**Reply by preetamslot** [reply]
@br41nslug Can you give an example of the rule?
I have tried:
```
{
	"_and": [
		{
			"$CURRENT_ROLE": {
				"_eq": "e0a89954-59e1-4e05-8413-30eb4626868c"
			}
		}
	]
}
```

but this does not work for me :(

**Reply by br41nslug** [reply]
Looks about right, in what context are you using this? and how have you determined it isnt working as expected?

**Reply by preetamslot** [reply]
I was using it as a condition to hide fields,
this was before I found out that I can hide fields in the permissions settings.

**Comment by hmaskat17** [comment]
Is it possible to disable a field but still display it in an item's details page for a specific user role? I have a Flow that updates the field in the background so the role needs update permissions, and read permissions to display the field. I did not see a way to get it to work in field > conditions filter.
